positionContract.sol

```diff
@@ -0,0 +1,255 @@
+pragma solidity ^0.5.2;
+import {CErc20, ComptrollerInterface, CToken} from "./CERC20.sol";
+import {CEther} from "./CEther.sol";
+import {UniswapExchangeInterface} from "./uniswap.sol";
+import {ERC20Interface} from "./ERC20.sol";
+
+
+contract positionContract {
+    // The user's address
+    address public ownerAddress; 
+    string public tradeType; 
+    string public asset;
+    
+    ERC20Interface token;
+    CErc20 cToken;
+    UniswapExchangeInterface tokenExchange;
+    
+    ERC20Interface collateral;
+    UniswapExchangeInterface collateralExchange; 
+    CErc20 cCollateral; 
+    
+    uint256 private tokenBalance; 
+    uint256 public amtShorted;
+    uint256 private borrowBalance;
+    uint256 private supplyBalance;
+    uint256 private collateralToSupply;
+    
+    address private factoryAddress;
+    
+       constructor (address owneraddr, 
+                string memory _asset, 
+                address _collateralAddr, 
+                address _cCollateralAddr, 
+                address _collateralExchangeAddr, 
+                address _tokenAddr, 
+                address _cTokenAddr, 
+                address _tokenExchangeAddr, 
+                string memory _tradeType) public{ 
+        // Ensures that the factory contract is the only one instantiating new shortContracts
+        factoryAddress = msg.sender;
+        ownerAddress = owneraddr;
+        // "s" for short, "l" for leverage
+        tradeType = _tradeType;
+        asset = _asset;
+        amtShorted = 0;
+        tokenBalance = 0;
+        
+        collateralExchange = UniswapExchangeInterface(_collateralExchangeAddr);
+        cCollateral = CErc20(_cCollateralAddr);
+        collateral = ERC20Interface(_collateralAddr);
+        token = ERC20Interface(_tokenAddr);
+        cToken = CErc20(_cTokenAddr);
+        tokenExchange = UniswapExchangeInterface(_tokenExchangeAddr);
+    }
+    
+    // This function transfers in the collateral
+    function transferInCollateral(uint256 collateralAmt) private {
+        require(msg.sender == ownerAddress); 
+        require(collateralAmt > 0);
+        
+        // TODO: ensure that approve happens in javascript before this is even called
+        collateral.transferFrom(msg.sender, address(this), collateralAmt);
+        
+    }
+    
+    // This function mints a new compound cCollateral token
+    function mintCollateral(uint256 amt) private {
+        require(amt > 0);
+        collateral.approve(address(cCollateral), amt); // approve the transfer
+        assert(cCollateral.mint(amt) == 0); // mint the cTokens and assert there is no error
+    }
+    
+     // This function borrows the token being shorted from compound
+    function borrowToken(uint256 amt) private {
+        require(amt > 0);
+        ComptrollerInterface troll = ComptrollerInterface(0x8d2A2836D44D6735a2F783E6418caEDb86DA58d8);
```

Comment:
I suggest using a constant here instead. I'm pretty sure solidity does not optimize stuff like this but I could be wrong.